go/vt/vttablet/tabletserver/gc/tablegc.go

```diff
@@ -254,55 +270,47 @@ func (collector *TableGC) operate(ctx context.Context) {
+			// find something new to do.
+			go tableCheckTicker.TickNow()
 		case <-tableCheckTicker.C:
-			{
-				log.Info("TableGC: tableCheckTicker")
-				if gcTables, err := collector.readTables(ctx); err != nil {
-					log.Errorf("TableGC: error while reading tables: %+v", err)
-				} else {
-					_ = collector.checkTables(ctx, gcTables, dropTablesChan, transitionRequestsChan)
-				}
+			log.Info("TableGC: tableCheckTicker")
+			if err := collector.readAndCheckTables(ctx, dropTablesChan, transitionRequestsChan); err != nil {
+				log.Error(err)
 			}
 		case <-purgeReentranceTicker.C:
-			{
-				// relay the request
-				go func() { purgeRequestsChan <- true }()
-			}
+			// relay the request
+			go func() { purgeRequestsChan <- true }()
 		case <-purgeRequestsChan:
-			{
-				go func() {
-					tableName, err := collector.purge(ctx)
-					if err != nil {
-						log.Errorf("TableGC: error purging table %s: %+v", tableName, err)
-						return
-					}
-					if tableName == "" {
-						// No table purged (or at least not to completion)
-						// Either because there _is_ nothing to purge, or because PURGE isn't a handled state
-						return
-					}
-					// The table has been purged! Let's move the table into the next phase:
-					_, _, uuid, _, _ := schema.AnalyzeGCTableName(tableName)
-					collector.submitTransitionRequest(ctx, transitionRequestsChan, schema.PurgeTableGCState, tableName, true, uuid)
-					collector.removePurgingTable(tableName)
-					// Chances are, there's more tables waiting to be purged. Let's speed things by
-					// requesting another purge, instead of waiting a full purgeReentranceInterval cycle
-					time.AfterFunc(time.Second, func() { purgeRequestsChan <- true })
-				}()
-			}
-		case dropTable := <-dropTablesChan:
-			{
-				log.Infof("TableGC: found %v in dropTablesChan", dropTable.tableName)
-				if err := collector.dropTable(ctx, dropTable.tableName, dropTable.isBaseTable); err != nil {
-					log.Errorf("TableGC: error dropping table %s: %+v", dropTable.tableName, err)
+			go func() {
+				tableName, err := collector.purge(ctx)
+				if err != nil {
+					log.Errorf("TableGC: error purging table %s: %+v", tableName, err)
+					return
+				}
+				if tableName == "" {
+					// No table purged (or at least not to completion)
+					// Either because there _is_ nothing to purge, or because PURGE isn't a handled state
+					return
 				}
+				// The table has been purged! Let's move the table into the next phase:
+				_, _, uuid, _, _ := schema.AnalyzeGCTableName(tableName)
+				collector.submitTransitionRequest(ctx, transitionRequestsChan, schema.PurgeTableGCState, tableName, true, uuid)
+				collector.removePurgingTable(tableName)
+				// Chances are, there's more tables waiting to be purged. Let's speed things by
+				// requesting another purge, instead of waiting a full purgeReentranceInterval cycle
+				purgeReentranceTicker.TickAfter(time.Second)
```

Comment:
now a variable. And you're right, it's a minimum; first and foremost because it ensures the table name signature, which is partly composed of a timestamp, gets a unique value over time.

To be fair, the purge logic is going to eventually disappear. It's relevant to `5.7`, and to old `8.0` versions. Eventually we'll retire this.